src/Parser.HelpColors.cs

```diff
@@ -0,0 +1,98 @@
+using CommandLine.Analysis;
+using CommandLine.Attributes;
+using CommandLine.Attributes.Advanced;
+using OutputColorizer;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using CommandLine.Colors;
+
+namespace CommandLine
+{
+    public static partial class Parser
+    {
+        public static class Colors
+        {
+            private static IColors s_helpColors;
+            private static object s_lockObj = new object();
+
+
+            public static IColors Get(ConsoleColor backgroundColor = (ConsoleColor)(-1))
+            {
+                if (s_helpColors != null)
+                {
+                    return s_helpColors;
+                }
+
+                // if the color is not specified, default to the current console's background color.
+                if (backgroundColor == (ConsoleColor)(-1))
+                {
+                    backgroundColor = Console.BackgroundColor;
+                }
+
+                // Get the color scheme.
+                lock (s_lockObj)
+                {
+                    if (s_helpColors == null)
+                    {
+                        s_helpColors = GetDefault(backgroundColor);
+                    }
+                }
+
+                return s_helpColors;
+            }
+
+            private static IColors GetDefault(ConsoleColor backgroundColor)
+            {
+                switch (backgroundColor)
+                {
+                    case ConsoleColor.Black:
```

Comment:
Couldn't you write this with just one return per color kind and then use the case cascade? i.e:

```cs
case ConsoleColor.Black:
case ConsoleColor.Blue:
case ConsoleColor.DarkBlue:
    return new DarkBackgroundColors();
...
```

Reply:
Good point! I will do that! Originally I think I had different colors for each one :).